### PR TITLE
Refactor Emotions Graph

### PR DESCRIPTION
**Pull request by michaelwehrley**
* Refactor `.drawMoodEmotionGraphs` js method
* Add basic specs
* Update rake file to include `Graph`

[#113779289]

### CODE REVIEW

**Comment by michaelwehrley** [comment]
@ericcf do you have any idea why these js specs may be failing?  They don't fail locally.

File: spec/dummy/spec/javascripts/think_feel_do_engine/feel/mood_spec.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      $jasmine_content.empty();
+    });
+    
+    describe('sc.calcStartOfDay', function() {
+      it('returns the correct date', function() {
+        var params = {
+          day: '2016-02-24'
+        };
+        expect(Number(sc.calcStartOfDay(params)))
+          .toBe(Number(new Date('2016-02-24 00:00')));
```
Comment (ericcf) [inline spec/dummy/spec/javascripts/think_feel_do_engine/feel/mood_spec.js]: This doesn't look like a valid ISO8601 date. You're probably using a newer version of PhantomJS locally from what's installed on Travis. Try `YYYY-MM-DDTHH:mm:ss.sssZ` format.

Comment (ericcf) [inline spec/dummy/spec/javascripts/think_feel_do_engine/feel/mood_spec.js]: This doesn't look like a valid ISO8601 date. You're probably using a newer version of PhantomJS locally from what's installed on Travis. Try `YYYY-MM-DDTHH:mm:ss.sssZ` format.

Comment (michaelwehrley) [inline spec/dummy/spec/javascripts/think_feel_do_engine/feel/mood_spec.js]: @ericcf you are the man.  Thanks!

File: app/assets/javascripts/think_feel_do_engine/feel/mood.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+(function () {
+  'use strict';
+
+  sc.calcStartOfDay = function(date) {
+    var UNIT = 'day';
```
Comment (eschlange) [inline app/assets/javascripts/think_feel_do_engine/feel/mood.js]: Is this var necessary?

**Comment by eschlange** [comment]
@michaelwehrley Nice abstraction of JS from the view. Specs look good. Aside from the one comment and build failure, I think its good to go.

File: app/assets/javascripts/think_feel_do_engine/feel/mood.js (hunk 2 above)
Comment (michaelwehrley) [inline app/assets/javascripts/think_feel_do_engine/feel/mood.js]: Not anymore.  A hold over from the past.  Thanks.

**Pull request closed without merging**